### PR TITLE
Unity protocol

### PR DESCRIPTION
**Pull request by semuserable**
closes https://github.com/getsentry/sentry-unity/issues/49

~WIP~

### CODE REVIEW

File: src/Sentry.Unity/UnityEventProcessor.cs [hunk 1]
```diff
@@ -30,104 +28,119 @@ public UnityEventProcessor(IApplication? application = null)
...
+            sdk.AddPackage(UnitySdkInfo.PackageName, UnitySdkInfo.Version);
+            sdk.Name = UnitySdkInfo.Name;
+            sdk.Version = UnitySdkInfo.Version;
+        }
 
-            @event.Contexts.App.StartTime = DateTimeOffset.UtcNow.AddSeconds(-Time.realtimeSinceStartup);
+        private static void PopulateApp(App app)
+        {
+            app.StartTime = DateTimeOffset.UtcNow
+                // NOTE: Time API requires main thread
```
Comment (bruno-garcia) [inline src/Sentry.Unity/UnityEventProcessor.cs:53]: Can we do a runtime check? We better not run this if we're not on the main thread.
Fair point this is unrelated to the PR but rather broken right now

File: src/Sentry.Unity/UnityEventProcessor.cs [hunk 2]
```diff
@@ -30,104 +28,119 @@ public UnityEventProcessor(IApplication? application = null)
...
+            device.DeviceType = SystemInfo.deviceType.ToString();
+            device.CpuDescription = SystemInfo.processorType;
+            device.Timezone = TimeZoneInfo.Local;
+            device.ProcessorCount = SystemInfo.processorCount;
+            device.SupportsVibration = SystemInfo.supportsVibration;
+            device.Name = SystemInfo.deviceName;
+
+            // This is the approximate amount of system memory in megabytes.
+            // This function is not supported on Windows Store Apps and will always return 0.
+            device.MemorySize = SystemInfo.systemMemorySize;
```
Comment (bruno-garcia) [inline src/Sentry.Unity/UnityEventProcessor.cs:79]: Good comment. What about we check if it's `0` and avoid adding it altogether?

**Review by bruno-garcia — COMMENTED** [review]
Looking pretty good for a draft btw

File: src/Sentry.Unity/UnityEventProcessor.cs (hunk 1 above)
Comment (semuserable) [reply src/Sentry.Unity/UnityEventProcessor.cs:53]: Check if we're on the main thread before doing that?

File: src/Sentry.Unity/UnityEventProcessor.cs (hunk 2 above)
Comment (semuserable) [reply src/Sentry.Unity/UnityEventProcessor.cs:82]: I noticed that it's used 2 times and we already have the logic to check against `0`, will remove that one altogether)

**Comment by semuserable** [comment]
I think we should probably add `SystemInfo.graphicsShaderLevel` on `IApplication` level so we can unit test the conversion logic.

**Comment by semuserable** [comment]
Ready to be reviewed.

cc @bruno-garcia @bitsandfoxes

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**